packages/layout/src/footnotes/getFootNotes.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * calculates the locations  of the footnotes within the given node array
+ * this is used to check if a footnote is actually inside a node after line breaking and splitting
+ * 
+ * @param {Array} nodes 
+ * @returns Array
+ */
+function calculateFootnoteLocations(nodes) {
+  let foot_notes = [];
+  let str = '';
+  for (const item of nodes) {
+    if (item.props && item.props.footNote) {
+      foot_notes.push({ loc: str.length, el: item });
+    }
+    if (item.type == 'TEXT_INSTANCE') {
```

Comment:
can you also check the impact of having images together with text nodes?
See [here](https://react-pdf.org/repl?code=3187b0760ce02e00408a057025803c450298c0bc300500943807cf805030c00f0022230080b6198531165d400a02180e659a004f003619b006f61622003a0046200099080beed3a72a005430a58d3c54a8a231c8016187928c009d54c0066a83128d9b3803f1801856d5cc25180077242873187f1e6028005a00072547184f0e4d2a007a5d7d7774ec8313314943392830b1753a306a0049467e0c348f4362c2b359247a815526cd085b606c00220cce86880c804764002b740c00465919f8be21de980ce2442439cc181b18111e1800591e3060731e4cfcdcad7c9816e353391e0470907b6253a43e040c0880e582a1d41a1b4a33c4a1d2e8607a1e4e3f506233180826d35dbcc962b3586cb6be3b000dc2e98082dcf46c0d86d41b08853cdaad578c21af08453c06c351ac3d1b32c72d56eb0456c9a995e009dc993a03198ac7601000dc6432000c5c050593f8f848681d8f0120e238788c2408884002e18001c800f210608f0444a4b4006838488b65bcc50283c420668c8651c6ab91f1a03c32b0164a04606426203b43a94194242c00cc190026801f5b40069001a800d840000e003a88109f2002a929538c2cc0034009c3c00209b764dec725ac8aa0552b409002abc703000329b5c7960c3ab80514c1ea388a1505bf59a78b589448301f1b4207885b5300565746eb73bbe00085d050102302d85d3d35374a6deee00125f24000bcd50ea3d9f4a15433c60328a0310d726883561c75fc300b400260005940ca1307d0db110fe3003d600584c16c175a0b55952344d734ad5b5ed4757b60340f793e5b0a0cd060a80e09fc1098016442d0b02292c270bc208bb188cd1ea5b0b53006f6f5ef0b5908001940902dd041e470320981d74e0d88e2b88584b3e224a922d1e2f8b62c8e354d0f5a8c4ce898054f422916338632775337892360f834cd429a0c2a0412f85c2ad19810680904708431274d23c8eb2ad6d13a3316200095ef0b9ecc7260544b8ed328372c07ccec08c00ee280d7278492772fd6c5fdff1114c852948e1b2cb1ac3b05cca1749f3b8cf3c4aaaa4992ef07c604425acd002a0a42cb5f0d6144be340110be0f4f87f1a2e5340cdc04000e4987913aad25f78cc2241c00f47879020100440f83018bbab54f48f2f8c5164d1b5309b383111c2802d2fb283aaf82f5febe2a6ec382e13e6a23169ba56ab4d68c036d6b5d7ed1532152ab0622e068654351606c5b0f02a076341f62d807200000)

Reply:
sure will check.